gateway-service/src/main/java/org/zowe/apiml/gateway/controllers/AuthController.java

```diff
@@ -89,32 +97,73 @@ public void invalidateJwtToken(HttpServletRequest request, HttpServletResponse r
     @DeleteMapping(path = ACCESS_TOKEN_REVOKE)
     @ResponseBody
     @HystrixCommand
-    public ResponseEntity<String> revokeAccessToken(@RequestBody() Map<String, String> token) throws Exception {
-        if (tokenProvider.isInvalidated(token.get(TOKEN_KEY), "null_service")) {
+    public ResponseEntity<String> revokeAccessToken(@RequestBody() Map<String, String> body) throws IOException {
+        if (tokenProvider.isInvalidated(body.get(TOKEN_KEY))) {
             return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
         }
-        tokenProvider.invalidateToken(token.get(TOKEN_KEY));
+        tokenProvider.invalidateToken(body.get(TOKEN_KEY));
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
-    @DeleteMapping(path = ACCESS_TOKEN_REVOKE + "/rules")
+    @DeleteMapping(path = ACCESS_TOKEN_REVOKE_MULTIPLE)
     @ResponseBody
     @HystrixCommand
-    public ResponseEntity<String> revokeAccessTokensWithRules(@RequestBody() RulesRequestModel rulesRequestModel) throws Exception {
-        String ruleId = rulesRequestModel.getRuleId();
-        long timeStamp = rulesRequestModel.getTimeStamp();
-        tokenProvider.invalidateTokensUsingRules(ruleId, timeStamp);
-        return new ResponseEntity<>(HttpStatus.OK);
+    public ResponseEntity<String> revokeAllUserAccessTokens(@RequestBody(required = false) RulesRequestModel rulesRequestModel) {
+        if (SecurityContextHolder.getContext().getAuthentication() == null) {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+        String userId = SecurityContextHolder.getContext().getAuthentication().getPrincipal().toString();
+        long timeStamp = 0;
+        if (rulesRequestModel != null) {
+            timeStamp = rulesRequestModel.getTimestamp();
+        }
+        tokenProvider.invalidateAllTokensForUser(userId, timeStamp);
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    @DeleteMapping(path = ACCESS_TOKEN_REVOKE_MULTIPLE + "/user")
+    @ResponseBody
+    @HystrixCommand
+    @PreAuthorize("hasSafServiceResourceAccess('SERVICES', 'READ')")
+    public ResponseEntity<String> revokeAccessTokensForUser(@RequestBody() RulesRequestModel requestModel) throws JsonProcessingException {
+        long timeStamp = requestModel.getTimestamp();
+        String userId = requestModel.getUserId();
+        if (userId == null) {
+            return badRequest();
+        }
+        tokenProvider.invalidateAllTokensForUser(userId, timeStamp);
+
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    ResponseEntity<String> badRequest() throws JsonProcessingException {
```

Comment:
Should be `private`, I would also move this down under all endpoints to get it out of the way. Should it be named `badRequest` or something that makes it clear it is a bad request only for revoking tokens?